genlib/bspnode.cpp

```diff
@@ -16,10 +16,106 @@
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "bspnode.h"
+#include <stack>
 
 // Binary Space Partition
 
-void BSPNode::make(Communicator *communicator, time_t atime, const prefvec<TaggedLine>& lines, BSPNode *par)
+/* Takes a set of lines and creates a binary-space-partition tree by starting from a
+ * root node, setting its left and right nodes and recursively doing the same process
+ * over those. Through this process the set of lines is split in two (one set for each
+ * left and right nodes) and those are split and passed again further down the recursion.
+ * While the original implementation was actually recursive it was hitting the recursion
+ * limit when the input was a large number of lines that fell on the same side (i.e. an
+ * arc divided in 500 pieces). It has been refactored here to an iterative solution, where
+ * the current node (left or right) is pushed to a stack along with the relevant set of lines.
+ */
+
+void BSPNode::make(Communicator *communicator, time_t atime, const std::vector<TaggedLine>& lines, BSPNode *par)
```

Comment:
Does this function need to be a member of BSPNode? It seems that the only time the `this` pointer is used it's for the initial push on the node stack and to call makeLines...
Ideally, this could be restructured so that BSPNode is mainly a data container that takes its line and tag in the constructor.